universum/modules/structure_handler.py

```diff
@@ -264,6 +265,20 @@ def execute_steps_recursively(self, parent: Step,
 
         return not some_step_failed
 
+
+    def execute_conditional_step(self, step, step_executor):
+        self.configs_current_number += 1
+        step_name = self._build_step_name(step.name)
+        conditional_step_succeeded = False
+        with self.block(block_name=step_name, pass_errors=True):
+            process = self.execute_one_step(step, step_executor);
+            conditional_step_succeeded = not process.get_error()
+        step_to_execute = step.if_succeeded if conditional_step_succeeded else step.if_failed
+        return self.execute_steps_recursively(parent=Step(),
+                                              children=Configuration([step_to_execute]),
```

Comment:
This assumes that only single step is supported in if_succeeded/if_failed parameters.
I propose to support whole Configuration instead.

Reply:
The "implement Configuration support instead of Step for branch steps" entry was added to the [task](https://github.com/Samsung/Universum/issues/709).